pyproject.toml

```diff
@@ -18,4 +19,189 @@ ignore_missing_imports = true
+    "ANN204",  # Missing type annotation for special method `__init__`
+    "ANN401",  # Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`
+    "E501",    # Line length too long
+    "E731",    # Do not assign a lambda expression, use a def
+    "E741",    # Ambiguous variable name
+    "G004",    # Logging statement uses f-string
+    "PLR0911", # Too many return statements
+    "PLR0912", # Too many branches
+    "PLR0913", # Too many arguments to function call
+    "PLR0915", # Too many statements
+    "PLW0603", # Using the global statement to update `<VAR>` is discouraged
+    "PT018",   # Assertion should be broken down into multiple parts
+    "Q000",    # Single quotes found but double quotes preferred
+    "S101",    # Use of assert detected
+    "SLF001",  # Private member accessed: `_preemptible`
+
+    "ARG001",  # Unused function argument
+    "ARG002",  # Unused method argument
+
+    "PLR2004", # Magic value used
+
+    "ANN001",
+    "ANN202",
+    "C408",
+    "TID252",
+    "RET504",
+    "ERA001",
+    "UP032",
+    "RUF100",
+    "ISC001",
+    "PIE804",
+    "F401",
+    "C901",
+    "W605",
+    "RET505",
+    "ANN003",
+    "RUF013",
+    "UP031",
+    "RUF010",
+    "B006",
+    "ANN002",
+    "B023",
+    "EXE001",
+    "G001",
+    "SIM108",
+    "RUF005",
+    "G002",
+    "PD901",
+    "N999",
+    "SIM118",
+    "SIM102",
+    "PLW2901",
+    "S603",
+    "ARG005",
+    "PGH003",
+    "B904",
+    "N802",
+    "ISC003",
+    "ANN205",
+    "S607",
+    "RUF015",
+    "E701",
+    "N818",
+    "PIE790",
+    "N803",
+    "A002",
+    "RUF012",
+    "W291",
+    "S113",
+    "S311",
+    "N806",
+    "PLR5501",
+    "F403",
+    "SIM115",
+    "B007",
+    "F841",
+    "C405",
+    "C419",
+    "SIM300",
+    "PD011",
+    "UP015",
+    "S602",
+    "Q002",
+    "ISC002",
+    "COM819",
+    "C416",
+    "DTZ005",
+    "G003",
+    "S608",
+    "PIE808",
+    "B008",
+    "S108",
+    "E402",
+    "S605",
+    "F821",
+    "RET507",
+    "RET503",
+    "UP030",
+    "UP026",
+    "PLR1714",
+    "C403",
+    "PLR1711",
+    "PIE810",
+    "DTZ011",
+    "S105",
+    "BLE001",
+    "C401",
+    "C400",
+    "PLR0402",
+    "SIM201",
+    "RET506",
+    "C417",
+    "PD010",
+    "PLW1510",
+    "A001",
+    "W292",
+    "PYI024",
+    "Q003",
+    "S301",
+    "RET501",
+    "PD003",
+    "SIM117",
+    "RUF002",
+    "UP027",
+    "SIM105",
+    "E713",
+    "S324",
+    "S310",
+    "Q001",
+    "UP020",
+    "S506",
+    "N805",
+    "E712",
+    "E401",
+    "SIM212",
+    "DTZ002",
+]
+
+[tool.ruff.lint.isort]
+
+section-order = ["future", "standard-library", "third-party", "hail", "cpg", "first-party", "local-folder"]
+
+[tool.ruff.lint.isort.sections]
+hail = [
+    "hail",
+    "hailtop",
+]
+# Adjust these for each repository, e.g., removing those that should be
+# local rather than CPG. Also fill in extend_skip below if there are any
```

Comment:
The `extend_skip` bit was for isort, so can be replaced by a reference to `tool.ruff.extend-exclude`.

But basically the whole comment is only really useful in the template directory, so you could just remove it.

Reply:
Yeah agreed, I've removed this comment